### ISSUE TITLE
Click on a variation of Walking Corpse or Soulless does not show the unit

### ISSUE THREAD

**Issue by Toranks** [wesnoth/wesnoth#7107]
### Game and System Information

- **Version:** 1.16
- **Downloaded from:** Steam
- **OS:** Windows 7

### Description of the bug

When clicking on the unit type in the UI, help windows says "You haven't discovered this unit yet", but navigating directly in the help to the unit brings up all the variations, even if I haven't discovered them. With the main variation (the basic zombie) this does not happen, show the Walking Corpse help correctly.

### Steps to reproduce the behavior

- Kill a unit with a different variation of walking corpse than the main
- Click on "Walking corpse" UI name on the right 

### Expected behavior

It should display the corresponding variance. And, if possible, say "Not yet discovered" for variations that REALLY have not yet been discovered, with the usual (?) symbol.

### Additional context

If i press "D" with unit selected, this error doesn't happen, show the variation correctly on help menu

**Comment by Wedge009** [comment]
#6011 may be related.

**Comment by soliton-** [comment]
Seems like it should be easy. Just need to use the same logic as the `Unit Description` functionality.

**Comment by Wedge009** [comment]
Does this still happen with gui2 Help in 1.19?

**Comment by sevu** [comment]
Yes. For a unit which is a variation, clicking the unit type in the sidebar ~~it now leaves now to titlescreen~~ it still doesn't open the help page the same way as when using the `d` hotkey,

**Comment by babaissarkar** [comment]
> Does this still happen with gui2 Help in 1.19?

Wasn't a frontend issue actually. The help browser was being fed the wrong help topic id (`unit_Walking Corpse` instead of `variation_Walking Corpse_rat`.)

**Comment by babaissarkar** [comment]
Can someone confirm the solution works? It may need to be backported.

**Comment by soliton-** [comment]
Works for me.

**Comment by sevu** [comment]
Works for WC, works not for Dunefolk lvl 0 unit in A New Land era. It fails there for the base unit too though, not only for variations.

With both units are lines such as `Failed to convert a string from utf8 to UCS4` in the log.
Before this change there were more such errors, text and sprite for the Dunefolk lvl 0 were previously not displayed.

**Comment by Wedge009** [comment]
> It fails there for the base unit too though, not only for variations.

I don't know the details, but this sounds like it might be an issue specific to the add-on...?

**Comment by soliton-** [comment]
When pressing d (Unit Type Description): `Showing topic: variation_ANLEra Dune Civilian_dune6: worker`

When clicking on the unit type report: `Mandatory widget ‘-variation_Dune Civilian_dune6’ hasn't been defined.`

**Comment by soliton-** [comment]
`u->type_id()` was erroneously changed to `u->type_name()`.

It would be even better if the exact same logic as in `help::show_unit_description` would be used. If it has to be a code copy then an explanatory comment should be added.

**Comment by babaissarkar** [comment]
> The utf8 thing is unrelated but might also be a bug.

This looks like the error:
```Failed to convert a string from utf8 to UCS4
2025-04-25 20:37:08 error gui/layout: pango_text::set_text text '​￼
Base unit:�' contains invalid utf-8, trimmed the invalid parts.```

**Comment by soliton-** [comment]
> > It would be even better if the exact same logic as in `help::show_unit_description` would be used. If it has to be a code copy then an explanatory comment should be added.
> 
> It'd actually require a slightly bigger adjustment to how clicking on a report works, that's why I stayed away from it. `help::show_unit_description` need an unit or unit_type, while the report mechanism is more generic: it works with any help topic id.

But why change the generic way of reports? This is about topic ID generation so I'd separate that out so it can be used in multiple places.
